docs/downstream-integrations/api/authentication.md

```diff
@@ -24,22 +24,133 @@ The AE token must be passed in the HTTP `Authorization` (case insensitive) reque
+
+These credentials are essential for initiating the OAuth2 flow and obtaining access tokens.
+
+## Understand Scopes
+
+Collab.Land API uses scopes to define the level of access and permissions requested by the client application. Scopes determine the resources and operations that the access token can be used for. It is crucial to understand the available scopes and select the appropriate ones based on your application's requirements.
+
+The following scopes are available:
+
+**account:read**
+
+- Read all of your wallet address you connected with collab.land and balances of them.
+- Read basic user information including user name, id , pfp.
+- See communities you are in and administrating
+
+**access-control:read**
+Read user roles in a certain community
+
+**asset-ownership:read**
+Check ownership for the given account for the given token assets
+
+**gm-pass:read**
+Read gmPASS address and DID info
+
+**read-vcs**
+Read all public verifiable credentials issued to you
+
+**request-vcreds**
+Issue verifiable credentials to your DID (gmPASS)
+
+**kudos:read**
+Read certain type public verifiable credentials issued to you
+
+## Login with Collab.Land (LWC) Flow
+
+To initiate the OAuth2 flow, direct users to the Collab.Land OAuth2 authorization endpoint:
+
+```bash
+https://api.collab.land/oauth2/authorize
+```
+
+Include the following parameters in the URL:
+
+- **`response_type`**: Set this to **`code`** for the authorization code flow or **`token`** for the implicit token flow.
+- **`client_id`**: The client ID obtained during the application registration process.
+- **`redirect_uri`**: The URI to which Collab.Land will redirect the user after authentication. Must be one of your registered redirect URIs. Register your localhost URIs for testing.
+- **`scope`**: The desired scopes requested by your application, separated by spaces.
+- **`state`**: A unique string generated by your application to protect against cross-site request forgery attacks.
+
+For example, to initiate the authorization code flow:
+
+```bash
+https://api.collab.land/oauth2/authorize?response_type=code&client_id=YOUR_CLIENT_ID&redirect_uri=YOUR_REDIRECT_URI&scope=scope1+scope2&state=UNIQUE_STATE_STRING
+```
+
+Or, for the implicit token flow:
+
+```bash
+https://api.collab.land/oauth2/authorize?response_type=token&client_id=YOUR_CLIENT_ID&redirect_uri=YOUR_REDIRECT_URI&scope=scope1+scope2&state=UNIQUE_STATE_STRING
+```
+
+Upon visiting the authorization endpoint, users will be prompted to log in with their Collab.Land account and grant the requested permissions to your application.
+
+## Obtain an Access Token
+
+### Authorization Code Flow
+
+If you initiated the authorization code flow, Collab.Land will redirect the user back to the **`redirect_uri`** specified in the authorization request, along with an authorization code. Your application must exchange this code for an access token by making a request to the Collab.Land OAuth2 token endpoint:
```

Comment:
Authorization code flow is preferred if the client application has a backend that can securely access the `client_secret`. The authorization code can only be used once to exchange for the access token.